scripts/release/branch_detection.py

```diff
@@ -0,0 +1,72 @@
+"""
+Branch detection and cache scoping utilities for Evergreen CI.
+
+This module provides functions to detect the current git branch and generate
+cache scopes for BuildKit remote cache in different environments (local development,
+Evergreen patch builds, Evergreen regular builds).
+"""
+
+import subprocess
+from typing import Optional
+
+
+def get_current_branch() -> Optional[str]:
+    """
+    Detect the current git branch for cache scoping.
+
+    In CI environments like Evergreen, git rev-parse --abbrev-ref HEAD returns
```

Comment:
[2025/09/01 19:58:12.138] #9 importing cache manifest from 268558157000.dkr.ecr.us-east-1.amazonaws.com/dev/cache/mongodb-kubernetes:master
[2025/09/01 19:58:12.138] #9 ERROR: failed to configure registry cache importer: 268558157000.dkr.ecr.us-east-1.amazonaws.com/dev/cache/mongodb-kubernetes:master: not found
[2025/09/01 19:58:12.138] #10 [builder  2/11] ADD https://github.com/stedolan/jq/releases/download/jq-1.8.1/jq-linux-amd64 /usr/local/bin/jq
[2025/09/01 19:58:12.322] #10 DONE 0.2s
[2025/09/01 19:58:12.322] #11 importing cache manifest from 268558157000.dkr.ecr.us-east-1.amazonaws.com/dev/cache/mongodb-kubernetes:add-caching
[2025/09/01 19:58:12.322] #11 inferred cache manifest type: application/vnd.oci.image.manifest.v1+json done